### PR TITLE
creates new structure for network docs

### PR DESCRIPTION
**Pull request by acozine**
##### SUMMARY
Creates a bare-bones landing page for the network docs at the top level of the documentation. This structural change will allow future work on network docs to move ahead more independently. 

##### ISSUE TYPE
 - Docs Pull Request

##### COMPONENT NAME
Network documentation

##### ANSIBLE VERSION
```
2.5
```

##### ADDITIONAL INFORMATION
See https://docs.google.com/document/d/1JzmxJn6DUPgQ-tjpgEqq-pjtTb4LbhRQicpBM28lCdI for the bigger picture.

### CODE REVIEW

**Comment by acozine** [comment]
@dharmabumstead since this PR removes the old `intro_network` page, I'd like to set a redirect from there to the new `network` page, so external links remain workable. I've heard you're doing a bunch of redirects as part of your refactoring work - can you add this one to the list? I'm happy to help when you get to that stage.

File: docs/docsite/rst/network.rst [hunk 1]
```diff
@@ -0,0 +1,15 @@
+Ansible Network
```
Comment (gundalow) [inline docs/docsite/rst/network.rst:1]:
```
***************
Ansible Network
***************
```

**Review by gundalow — COMMENTED** [review]
Just a minor thing

File: docs/docsite/rst/network.rst (hunk 1 above)
Comment (acozine) [reply docs/docsite/rst/network.rst:1]: Current docs have inconsistent header styling (it has probably accumulated through the years). I'd prefer to discuss this, come up with a consistent notation, and submit one PR that makes all the docs use the same notation.

Comment (acozine) [reply docs/docsite/rst/network.rst:1]: Created https://github.com/ansible/ansible/issues/35520 to address this.

Comment (gundalow) [reply docs/docsite/rst/network.rst:1]: Agreed. Ignore this comment and we can address this in the bulk fix up PR.

As long as Shippable RST check passes  (which it is) I'm happy.

**Review by gundalow — APPROVED** [review]

**Review by dharmabumstead — CHANGES_REQUESTED** [review]
This is a great start! Nicely done, @acozine!

**Review by dharmabumstead — APPROVED** [review]

**Pull request merged**